java/core/src/main/scala/io/pdal/PointCloud.scala

```diff
@@ -0,0 +1,101 @@
+package io.pdal
+
+import java.nio.ByteBuffer
+import java.util
+
+import scala.collection.JavaConversions._
+
+/**
+  * PointCloud abstraction to work with packed point(s) in JVM memory.
+  * SizedDimType contains size and offset for a particular packed point with the current set of dims.
+  **/
+case class PointCloud(bytes: Array[Byte], dimTypes: util.Map[String, SizedDimType], metadata: String = "", schema: String = "") {
```

Comment:
`metadata` and `schema` shouldn't be on here. The class doesn't use it and it's really a separate concern from this class. Seems like it's here to substitute for a composition, (metadata, schema, pointCloud)